superbench/benchmarks/micro_benchmarks/memory_bw_performance.py

```diff
@@ -0,0 +1,148 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+"""Module of the Cuda memory performance benchmarks."""
+
+import os
+
+from superbench.common.utils import logger
+from superbench.benchmarks import BenchmarkRegistry, Platform, ReturnCode
+from superbench.benchmarks.micro_benchmarks import MicroBenchmarkWithInvoke
+
+
+class MemBwCuda(MicroBenchmarkWithInvoke):
+    """The Cuda memory bus bandwidth performance benchmark class."""
+    def __init__(self, name, parameters=''):
+        """Constructor.
+
+        Args:
+            name (str): benchmark name.
+            parameters (str): benchmark parameters.
+        """
+        super().__init__(name, parameters)
+
+        self._bin_name = 'bandwidthTest'
+        self.__mem_types = ['htod', 'dtoh', 'dtod']
+        self.__mode = ['shmoo', 'range']
```

Comment:
seems you cannot test `--mode range` because it requires additional range parameters
maybe just expose a `--shmoo` store_true arg?

Reply:
changed, thanks for pointing the issue